### PR TITLE
feat(pipelines): ECR source action 

### LINKED ISSUE #16378: (pipelines): Add support for Amazon ECR source action
Add [Amazon ECR](https://docs.aws.amazon.com/codepipeline/latest/userguide/integrations-action-type.html#integrations-source-ecr) source action to [pipelines.CodePipelineSource](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_pipelines.CodePipelineSource.html) class.

### Use Case

Pipelines that trigger on push to Amazon ECR repository (e.g., base image update due to security CVE)

### Proposed Solution

Add `pipelines.CodePipelineSource.ecr(...)` factory method.

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by alexpulver**
Closes #16378 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by alexpulver** [comment]
I will need to rework the `actionName` definition, because `/` can appear in ECR repository name (namespace), and is not a valid character in CodePipeline action name.

**Review by rix0rrr — APPROVED** [review]

**Comment by alexpulver** [comment]
@rix0rrr I changed the code to handle the `/` namespace character by replacing it with `_` character. I tested manually that `const formattedRepositoryName = Fn.join('_', Fn.split('/', this.repository.repositoryName));` works on repositories with and without namespace, because I don't think it's possible to test without deploying the stack. Below are the snippets from the template I used for manual testing. Can you please let me know if it's acceptable from your perspective?

No namespace:
```json
{
  "Resources": {
    "Repository22E53BBD": {
      "Type": "AWS::ECR::Repository"
    },
<snip>
```

With namespace:
```json
{
  "Resources": {
    "Repository22E53BBD": {
      "Type": "AWS::ECR::Repository",
      "Properties": {
        "RepositoryName": "namespace/repo"
      }
    },
<snip>
```

Pipeline:
```json
{
  "Resources": {
    <snip>
    "Pipeline9850B417": {
      "Type": "AWS::CodePipeline::Pipeline",
      "Properties": {
        "RoleArn": {
          "Fn::GetAtt": [
            "PipelineRoleB27FAA37",
            "Arn"
          ]
        },
        "Stages": [
          {
            "Actions": [
              {
                "InputArtifacts": [],
                "ActionTypeId": {
                  "Version": "1",
                  "Owner": "AWS",
                  "Category": "Source",
                  "Provider": "ECR"
                },
                "OutputArtifacts": [
                  {
                    "Name": "Source"
                  }
                ],
                "RunOrder": 1,
                "Configuration": {
                  "ImageTag": "latest",
                  "RepositoryName": {
                    "Ref": "Repository22E53BBD"
                  }
                },
                "Name": {
                  "Fn::Join": [
                    "_",
                    {
                      "Fn::Split": [
                        "/",
                        {
                          "Ref": "Repository22E53BBD"
                        }
                      ]
                    }
                  ]
                }
              }
            ],
            "Name": "Source"
          },
<snip>
```

**Comment by alexpulver** [comment]
@rix0rrr would be glad to have your input on the above

**Review by rix0rrr — DISMISSED** [review]

File: packages/@aws-cdk/pipelines/lib/codepipeline/codepipeline-source.ts [hunk 1]
```diff
@@ -51,16 +52,37 @@ export abstract class CodePipelineSource extends Step implements ICodePipelineAc
    * Example:
    *
    * ```ts
-   * const bucket: IBucket = ...
-   * CodePipelineSource.s3(bucket, {
-   *   key: 'path/to/file.zip',
+   * import * as ecr from '@aws-cdk/aws-ecr';
```
Comment (npvisual) [inline packages/@aws-cdk/pipelines/lib/codepipeline/codepipeline-source.ts:55]: @alexpulver  : should the following comments have been modified ? I think they still apply to the `s3` factory method.

**Review by rix0rrr — APPROVED** [review]

**Pull request merged**